tests/kernel/mp/src/main.c

```diff
@@ -93,14 +98,18 @@ FUNC_NORETURN void cpu1_fn(void *arg)
  */
 ZTEST(multiprocessing, test_mp_start)
 {
-	cpu_arg = 12345;
+	for (int i = 1; i < CONFIG_MP_MAX_NUM_CPUS; i++) {
+		TC_PRINT("Starting CPU #%d...\n", i);
 
-	arch_start_cpu(1, cpu1_stack, CPU1_STACK_SIZE, cpu1_fn, &cpu_arg);
+		cpu_arg = 12345 * i;
 
-	while (!cpu_running) {
-	}
+		arch_start_cpu(i, cpu_stacks[i], CPU_STACK_SIZE, cpu_fn, &cpu_arg);
 
-	zassert_true(cpu_running, "cpu1 didn't start");
+		while (!cpu_running[i]) {
+		}
+
+		zassert_true(cpu_running[i], "cpu #%d didn't start", i);
```

Comment:
Can we put a time bound on this while-loop? Otherwise the error message does not make much sense.

Reply:
Added code to wait for 5 seconds for CPUs to start. The wait may be too long but we bail early if CPU has started. This should give enough time for odd architectures (if any) to start another CPU.